src/controllers/searchVideos.controller.js

```diff
@@ -11,7 +12,12 @@ exports.searchVideos = async function (req, res, next) {
   }
 
   try {
-    const ranks = await fetchVideosRanks(userQuery);
+    const recommendedSearchKeyword = await checkUserInputSpelling(userQuery);
+    const correctedInput = shouldCheckSpell
+      ? await checkUserInputSpelling(userQuery)
+      : userQuery;
```

Comment:
```suggestion
    const recommendedSearchKeyword = await checkUserInputSpelling(userQuery);
    const correctedInput = shouldCheckSpell
      ? recommendedSearchKeyword
      : userQuery;
```
중복된 계산을 하나로 줄일 수 있을 것 같습니다!

Reply:
네! 수정하겠습니다!